NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Native/IFbClient.cs

```diff
@@ -269,4 +351,98 @@ public interface IFbClient
 
 		#endregion
 	}
+
+	internal static class IFbClientEx
+	{
+		public static IntPtr isc_dsql_allocate_statement_ex(
```

Comment:
Take a look at IFbClient, I've changed prototype and marshalling of a handle.
And I've already implemented injection of IFbClient instance using IL gen.